docs/design/direct-assign-volume.md

```diff
@@ -0,0 +1,139 @@
+# Background 
+
+Today, filesystem volumes are passed in to the guest VM utilizing virtio-fs. In a typical flow, a CSI controller
+will make sure backing storage is attached to the node (often a block device), and then a CSI node driver will
+create the filesystem (if needed) and mount it.
+
+![Volume handling without direct-assignment](arch-images/DA-not-enabled.png)
+
+Ideally, for a VM, we will consume the block device itself and not utilize the host mount. This simplifies volume management
+on the host, and is more efficient from a storage IO perspective. 
+
+![Volume handling with direct-assignment](arch-images/DA-UC.png)
+
+# Proposal - Introduce pattern for directly assigning block device for mounting in the guest
+
+In the existing upstream, there is no direct communication between CSI and the container runtime. By the
+time volume information is presented to the runtime, it is a path on the given host filesystem. We need a method to:
+ - enlighten the CSI driver that a given PVC is to be utilized by Kata (or any other runtime that would prefer to mount the volume on their own).
+ - communication path between the CSI node driver and the runtime to detail: what the backing block device is,
+   what kind of filesystem it is, and where it should be mounted with respect to the container rootfs.
+
+Ideally we'd be able to come up with a solution that would facilitate a runtime calling into CSI instead of Kubelet, as is done today for CNI. Working within the existing constraints that exist, we are proposing a pattern that can be used to facilitate direct-volume assignment, while working with other upstream projects for a more direct method. 
```

Comment:
- Line too long.
- s/the existing constraints that exist/the existing constraints/